### PR TITLE
I47 fix CSV deserialization of Union{Nulltype,AnyType}

### LINKED ISSUE #47: Compatible with CSV deserialization of Union{Nothing, Another} type
I noticed that when deserializing the CSV, missing values ​​were assigned the value `missing`. Of course, this is completely fine and complies with the Julia specification.

But for fields with Union{Nothing, Another} definition, deserialization will report an error:

``` julia
struct Data
           id::Int
           name::Union{String,Nothing}
end
csv = Serde.to_csv([Data(1,nothing),Data(2,"name2")])
# "id,name\n1,\n2,name2\n"
Serde.deser_csv(Data,csv)
#ERROR: WrongType: for 'Data' value 'missing' has wrong type 'name::Missing', must be 'name::Union{Nothing, String}'
```

Should we need to compatible with the Nothing type during deserialization, so that fields of type Union{Nothing, Another} can also be successfully deserialized?

I don't know if encouraging such compatibility is a good behavior, but I think that since the type Union{Nothing, Another} field is allowed to be serialized, we should support deserialization of this type, otherwise support of field Union{Nothing, Another} should be removed or throw error when serializing.

### PR DESCRIPTION
**Pull request by RongkunWang**
First attempt to contribute to a Julia package. Fix #47.

Used the code in the issue to create a test. 

### Pull request checklist

- [x] Did you bump the project version?
- [x] Did you add a description to the Pull Request?
- [x] Did you add new tests?
- [x] Did you add reviewers?

### CODE REVIEW

File: src/De/Deser.jl [hunk 1]
```diff
@@ -540,6 +540,9 @@ function deser(::CustomType, ::Type{D}, data::N)::D where {D<:Any,N<:NamedTuple}
         val = get(data, key, default_value(D, Val(name)))
         val = isnothing(val) ? nulltype(type) : val
         val = isempty(D, val) ? nulltype(type) : val
+        if Nothing <: type && !(Missing <: type) && ismissing(val)
+            val = nothing
+        end
```
Comment (gryumov) [inline src/De/Deser.jl:545]: maybe? 

```suggestion
        val = isnothing(val) || ismissing(val) || isempty(D, val) ? nulltype(type) : val
```

**Comment by gryumov** [comment]
@RongkunWang Hi, thanks for the PR! Looks good, but can you handle this case too?

```julia
using Serde

j = """
    {
        "z":1,
        "x": null
    }
"""

struct DataN
    z::Int
    x::Union{String,Nothing}
end

struct DataM
    z::Int
    x::Union{String,Missing}
end

julia> Serde.deser(DataN, Serde.parse_json(j, null = nothing))
DataN(1, nothing)

julia> Serde.deser(DataN, Serde.parse_json(j, null = missing))
ERROR: WrongType: for 'DataN' value 'missing' has wrong type 'x::Missing', must be 'x::Union{Nothing, String}'

julia> Serde.deser(DataM, Serde.parse_json(j, null = nothing))
DataM(1, missing)

julia> Serde.deser(DataM, Serde.parse_json(j, null = missing))
DataM(1, missing)
```

File: test/deser.jl [hunk 2]
```diff
@@ -664,4 +664,20 @@ using Test, Dates
...
+
+      #  csv = """
+      #  "id","name"
+      #  1,
+      #  2,"name2"
+      #  """;
+      csv = Serde.to_csv([Data(1, nothing), Data(2, "name2")])
+      exp_obj =      Data[Data(1, nothing), Data(2, "name2")]
+      @test Serde.deser_csv(Data, csv) == exp_obj
+    end
```
Comment (gryumov) [inline test/deser.jl:682]:
```suggestion

@testset "Case №38: Deserialization with missing field with Union{Nulltype,AnyType}" begin
    struct WithNothing
        y::Int64
        x::Union{String,Nothing}
    end

    @test Serde.deser(WithNothing, Dict("y" => 1, "x" => nothing)) == WithNothing(1, nothing)
    @test Serde.deser(WithNothing, Dict("y" => 2, "x" => missing)) == WithNothing(2, nothing)

    struct WithMissing
        y::Int64
        x::Union{String,Missing}
    end

    @test Serde.deser(WithMissing, Dict("y" => 3, "x" => nothing)) == WithMissing(3, missing)
    @test Serde.deser(WithMissing, Dict("y" => 4, "x" => missing)) == WithMissing(4, missing)
end
```

**Comment by RongkunWang** [comment]
Hi @gryumov ,

I see, the `nulltype()` will always be the correct type from the union so I should have just used that with your suggestion.

So now I'm applying the same technique for the other dispatch of `deser`.

**Review by gryumov — APPROVED** [review]

**Comment by gryumov** [comment]
@RongkunWang, thanks for the bug fix

**Pull request merged**